data-prepper-plugins/log-http-source/src/main/java/com/amazon/dataprepper/plugins/source/loghttp/LogThrottlingRejectHandler.java

```diff
@@ -0,0 +1,39 @@
+/*
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  The OpenSearch Contributors require contributions made to
+ *  this file be licensed under the Apache-2.0 license or a
+ *  compatible open source license.
+ *
+ *  Modifications Copyright OpenSearch Contributors. See
+ *  GitHub history for details.
+ */
+
+package com.amazon.dataprepper.plugins.source.loghttp;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.server.Service;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.throttling.ThrottlingRejectHandler;
+
+import javax.annotation.Nullable;
+
+public class LogThrottlingRejectHandler implements ThrottlingRejectHandler<HttpRequest, HttpResponse> {
+    private final int maxPendingRequests;
+
+    public LogThrottlingRejectHandler(int maxPendingRequests) {
+        this.maxPendingRequests = maxPendingRequests;
```

Comment:
Add validation here.

Reply:
I will add validation in the constructor of LogHTTPSourceConfig since that is the entry point for input

Reply:
It's good to have validation here if we plan to decouple with LogHTTPSourceConfig or use this with other class.

Reply:
This will create duplicate checks throughout our repository. I am all for being defensive but redundant code is hard to maintain. I think we need to come to an agreement and define expectations as a group.

Reply:
> This will create duplicate checks throughout our repository. I am all for being defensive but redundant code is hard to maintain. I think we need to come to an agreement and define expectations as a group.

+1.